text/0002-bundle-add-and-remove.md

```diff
@@ -0,0 +1,275 @@
+  gem "minitest-reporters" "~> 1.1"
+end
+```
+
+### Description
+
+Instead of adding gems to the Gemfile in the format
+
+```ruby
+gem "minitest", "~> 5.11", group: [:test]
+```
+
+the `add` command works with group blocks
+
+
+```ruby
+group :test do
+  gem "minitest", "~> 5.11"
+end
+```
+
+which are more natural for users.
+
+It appends gems to the end of the group block if the group already exists. It also supports adding to multiple groups.
+
+## 2. adding multiple gems with options
+
+#### Proposed command
+
+```shell
+bundle add sinatra:2.0 minitest:~>5.11
+```
+
+### Description
+
+The `add` command supports specifying versions for multiple gems.
+
+## 3. adding gems with optimistic or strict version ranges
+
+### 3.1 adding gems with strict version
+
+#### Proposed commands
+
+```shell
+bundle add sinatra minitest --strict
+```
+
+```ruby
+# Gemfile
+gem "sinatra", "2.0.0"
+gem "minitest", "5.11.3"
+```
+
+### 3.2 adding gems with optimistic version range
+
+#### Proposed commands
+
+```shell
+bundle add sinatra minitest --optimistic
+```
+
+```ruby
+# Gemfile
+gem "sinatra", ">= 2.0.0"
+gem "minitest", ">= 5.11.3"
+```
+
+### 3.3 adding gems without the version range
+
+```shell
+bundle add sinatra minitest --no-version
+```
+
+```ruby
+# Gemfile
+gem "sinatra"
+gem "minitest"
+```
+
+### Description
+
+The `add` command supports specifying different version ranges.
+
+## 4. skipping install step
+
+#### Proposed commands
+
+```shell
+bundle add sinatra minitest --skip-install
+```
+
+### Description
+
+The `--skip-install` flag skips the install step and just adds the gem to Gemfile.
+
+## 5. removing gems
+
+#### Proposed commands
+
+```ruby
+# Gemfile before
+source "https://rubygems.org"
+
+group :test do
+  gem "minitest"
+  gem "minitest-reporters"
+end
+```
+
+```shell
+bundle remove minitest minitest-reporters
+bundle rm minitest minitest-reporters
+```
+
+```ruby
+# Gemfile after
+source "https://rubygems.org"
+```
+
+### Description
+
+The `remove` command removes gems from the Gemfile, along with their corresponding group block if it remains empty after the deletion. By default it runs `bundle install` afterwards, which can also be skipped with the `--skip-install` flag.
+
+
+## 6. print flag
+
+#### Proposed commands
+
+```shell
+bundle add sinatra --print
+```
+
+Prints
+
+```
+sinatra, "~> 2.0"
+```
+
+#### Description
+
+The `--print` flag causes the gem name with the version to be printed out to STDOUT. It doesn't add the gem to the Gemfile and skips the install step.
+
+This is useful when editing a Gemfile with the editor since it can be used to insert the latest version of a gem to a cursor position.
+
+## 7. error messages
+
+### 7.1. adding gem when it already exists in the Gemfile
+
+```ruby
+# Gemfile
+source "https://rubygems.org"
+
+gem "minitest"
+```
+
+```
+bundle add minitest sinatra
+```
+
+Prints out and exits with error status code:
+
+```shell
+Gem "minitest" already present in the Gemfile. Aborting.
```

Comment:
I'd suggest changing `aborting` to `skipping`. Aborting sounds very serious and I wouldn't want it to abort in multi gem additions. I prefer it skips and adds remaining gems